packages/core/src/zip/select.js

```diff
@@ -0,0 +1,155 @@
+'use strict'
+
+import I from 'immutable'
+import R from 'ramda'
+
+import { isOfKind } from '../data'
+
+export const isStringArray = obj =>
+  R.is(Array)(obj) && (R.all(R.is(String))(obj) || obj.length === 0)
+const isImm = I.Iterable.isIterable
+const areImm = (...vals) => R.all(isImm)(vals)
+const isTrue = R.allPass([R.is(Boolean), R.equals(true)])
+const isLocation = loc =>
+  loc &&
+  loc.meta &&
+  loc.meta.isBranch &&
+  loc.meta.getChildren &&
+  loc.meta.makeItem
+export const isLocationArray = obj =>
+  R.is(Array)(obj) && R.not(R.isEmpty(obj)) && R.all(isLocation)(obj)
+
+// child :: String -> Location -> Location
+export const child = key => loc => {
+  if (loc.canGoDown()) {
+    let childLoc = loc.down()
+    const child = childLoc.value()
+    if (child.get('propertyName') === key && child.get('isSingle')) {
+      return childLoc.down()
+    }
+    while (childLoc.canGoRight()) {
+      childLoc = childLoc.right()
+      const child = childLoc.value()
+      if (child.get('propertyName') === key && child.get('isSingle')) {
+        return childLoc.down()
+      }
+    }
+  }
+  return null
+}
+
+// children :: (...keys) -> Location -> [Location]
+export const children = (...keys) => loc => {
```

Comment:
Same here. Do we need multiple "keys". `[...children('key1', loc), ...children('key2', loc)]` is enough.

Reply:
Alternatively a single arg  `key` could be interpreted as a string or array of strings.

Reply:
Both of the proposals don't handle the default case if nothing is provided. I introduced `childrenFor` that operates like the second thing you described, and `children` without keys.